File: src/model_v4/entities/SBS_HWSHIFT.ts
import { Column, Entity, Index, JoinColumn, ManyToOne, PrimaryColumn } from "typeorm";
import { SBS_HW } from "./SBS_HW";


@Entity("SBS_HWSHIFT",{schema:"sbsdb"})
@Index("sbshwshift_index3",["DIRECTION",])
@Index("sbshwshift_index2",["HOST",])
@Index("sbshwshift_index1",["SHIFTDATE",])
@Index("FK59BE7F58C79F763B",["hW_INDEX",])
export class SBS_HWSHIFT {

    @PrimaryColumn("bigint",{ 
        nullable:false,
        primary:true,
        name:"HWSHIFT_INDEX"
        })
    HWSHIFT_INDEX:string;
        

    @Column("bigint",{ 
        nullable:false,
        name:"AP_INDEX"
        })
    AP_INDEX:string;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"BETRIEBSSTELLE"
        })
    BETRIEBSSTELLE:string | null;
        

    @Column("varchar",{ 
        nullable:true,
        length:55,
        default: () => "'NULL'",
        name:"BEZEICHNUNG"
        })
    BEZEICHNUNG:string | null;
        

    @Column("varchar",{ 
        nullable:false,
        length:2,
        name:"DIRECTION"
        })
    DIRECTION:string;
        

    @Column("varchar",{ 
        nullable:true,
        length:50,
        default: () => "'NULL'",
        name:"HOST"
        })
    HOST:string | null;
        

    @Column("timestamp",{ 
        nullable:false,
        default: () => "'current_timestamp()'",
        name:"SHIFTDATE"
        })
    SHIFTDATE:Date;
        

   
    @ManyToOne(type=>SBS_HW, SBS_HW=>SBS_HW.sBS_HWSHIFTs,{ onDelete: 'RESTRICT',onUpdate: 'RESTRICT' })
    @JoinColumn({ name:'HW_INDEX'})
    hW_INDEX:SBS_HW | null;

}

File: SbsdbServer/sbsdbv4/model/SbsHwshift.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace hb.SbsdbServer.sbsdbv4.model {
    [Table("SBS_HWSHIFT")]
    public class SbsHwshift {
        [Column("HWSHIFT_INDEX", TypeName = "bigint(20)")]
        public long HwshiftIndex { get; set; }

        [Column("AP_INDEX", TypeName = "bigint(20)")]
        public long ApIndex { get; set; }

        [Column("BETRIEBSSTELLE", TypeName = "varchar(50)")]
        public string Betriebsstelle { get; set; }

        [Column("BEZEICHNUNG", TypeName = "varchar(55)")]
        public string Bezeichnung { get; set; }

        [Required]
        [Column("DIRECTION", TypeName = "varchar(2)")]
        public string Direction { get; set; }

        [Column("HOST", TypeName = "varchar(50)")]
        public string Host { get; set; }

        [Column("SHIFTDATE", TypeName = "timestamp")]
        public DateTime Shiftdate { get; set; }

        [Column("HW_INDEX", TypeName = "bigint(20)")]
        public long? HwIndex { get; set; }

        [ForeignKey("HwIndex")]
        [InverseProperty("SbsHwshift")]
        public virtual SbsHw HwIndexNavigation { get; set; }
    }
}
